tests/test_sycl_queue.py

```diff
@@ -1087,43 +1087,71 @@ def test_eig(device):
     assert_sycl_queue_equal(dpnp_vec_queue, expected_queue)
 
 
-@pytest.mark.usefixtures("allow_fall_back_on_numpy")
+@pytest.mark.parametrize(
+    "func",
+    [
+        "eigh",
+        "eigvalsh",
+    ],
+)
+@pytest.mark.parametrize(
+    "shape",
+    [
+        (4, 4),
+        (0, 0),
+        (2, 3, 3),
+        (0, 2, 2),
+        (1, 0, 0),
+    ],
+    ids=[
+        "(4, 4)",
+        "(0, 0)",
+        "(2, 3, 3)",
+        "(0, 2, 2)",
+        "(1, 0, 0)",
+    ],
+)
 @pytest.mark.parametrize(
     "device",
     valid_devices,
     ids=[device.filter_string for device in valid_devices],
 )
-def test_eigh(device):
-    size = 4
+def test_eigenvalue_symm(func, shape, device):
     dtype = dpnp.default_float_type(device)
-    a = numpy.arange(size * size, dtype=dtype).reshape((size, size))
-    symm_orig = (
-        numpy.tril(a)
-        + numpy.tril(a, -1).T
-        + numpy.diag(numpy.full((size,), size * size, dtype=dtype))
-    )
-    numpy_data = symm_orig
-    dpnp_symm_orig = dpnp.array(numpy_data, device=device)
-    dpnp_data = dpnp_symm_orig
+    numpy.random.seed(81)
+    a = numpy.random.randn(*shape).astype(dtype)
+    if a.size > 0:
+        if a.ndim > 2:
+            for i in range(a.shape[0]):
+                a[i] = numpy.conj(a[i].T) @ a[i]
+        else:
+            a = numpy.conj(a.T) @ a
```

Comment:
Looks as a common function which might be reused in other place to build Hermitian matrix. What do you think?

Reply:
I added a new `get_symm_herm_numpy_array` function to `helper.py`